### ISSUE TITLE
[V2] Fails to auto-install packages when using yarn workspaces.

### ISSUE THREAD

**Issue by SantoJambit** [parcel-bundler/parcel#5854]
# 🐛 bug report

When you have a monorepo with yarn set up, parcel tries to auto-install dependencies into the root package.json.
This even happens when I run parcel from the package folder.

## 🎛 Configuration (.babelrc, package.json, cli command)

Aside from the workspaces configuration:

```json
{
...
  "workspaces": {
    "packages": [
      "packages/*"
    ],
    "nohoist": [
      "**/parcel",
      "**/parcel/**",
      "**/@parcel/**"
    ]
  },
...
}
```

There is only one .parcelrc file
```
{
  "extends": "@parcel/config-default",
    "transformers": {
      "*.{jpg,png,svg}": [
        "@parcel/transformer-raw"
      ]
    }
}
```

## 🤔 Expected Behavior

It should work with monorepos.

## 😯 Current Behavior

```
Installing @parcel/transformer-posthtml...
@parcel/package-manager: yarn: Running this command will add the dependency to the workspace root rather than the workspace itself, which might not be what you want - if you
```

## 🔦 Context

I have a simple monorepo with one backend and one frontend package. The frontend package has parcel as dependency.
I was on 2.0.0-nightly.506 before, where nothing needed to be installed by parcel, so I never had the issue then.

I tried manually installing the items, but it still detects them as missing and tries to auto-install them again.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-nightly.596
| Node             | 15.8.0
| npm/Yarn         | yarn 1.22.4
| Operating System | Ubuntu 18.04

**Comment by mischnic** [comment]
This should probably not be the project root (= workspace root)

https://github.com/parcel-bundler/parcel/blob/36035d8d0d7f785771dd24207fe17d67ed85f269/packages/core/core/src/loadParcelPlugin.js#L75-L76

This would also explain why the resolution fails after installing manually

**Comment by SantoJambit** [comment]
Wow, that was a quick response!

**Comment by jeffpeck10x** [comment]
This seems to still be happening on `2.0.0-nightly.607+d77a5647`

I downgraded to the last version that I know that this was not an issue - `2.0.0-nightly.573` (i.e. it might be working on later version, but this was the last version that was working for me prior to updating to the latest)

**Comment by mischnic** [comment]
So the workaround here is installing Parcel in the monorepo root.

The problem is that isn't a simple solution for determining what to use instead of the project root here: https://github.com/parcel-bundler/parcel/issues/5854#issuecomment-779955980

**Comment by SantoJambit** [comment]
Wouldn't it be possible to check the package.json dependencies if it has parcel installed and if so, install the dependencies there?

**Comment by jeffpeck10x** [comment]
The latest @nightly, even with parcel in the monorepo root, does not work. The only current solution for me is staying `parcel@2.0.0-nightly.573`

**Comment by mischnic** [comment]
I meant:
1. Install `parcel` in the root
2. run Parcel with `--no-autoinstall`
3. Install all of the requested packages in the root

All other solutions might break due to hoisting or because Yarn 2 enforces declared dependencies

**Comment by toastal** [comment]
Like @jeffpeck10x, I've been stuck on an old version for quite some time using Yarn v1 workspaces. The version that works for me is slightly newer @ `2.0.0-nightly.579`. I haven't tried moving parcel up to the root, but this isn't a fix I want to do as I want the various packages's dependencies siloed. I tried upgrading to the latest nightly today, but it's still not working. I do have a `.parcelrc`(.json) file in the root of the problematic workspace (other workspaces are using Webpack).

**Comment by caillou** [comment]
I am currently running into this problem. My setup:

```
|- package.json // "workspaces": { 
|                                  "packages": ["p/*"],
|                                  "nohoist": ["**/*"]
|                                }
| p/
  |- foo
     | - package.json // "devDependencies": {
                                              "parcel": "^2.0.0-beta.2",
                                              "@parcel/transformer-sass": "2.0.0-beta.2"
                                            }
```

Running `parcel` throws the following:

```
Error: Failed to install @parcel/transformer-sass: Yarn failed to install modules:Child process failed
```

Running `parcel --no-autoinstall` throws the following:

```
@parcel/core: Cannot find Parcel plugin "@parcel/transformer-sass"
```

Installing in the root isn't an option, due to dependency problems with regard to `eslint` and different `typescript` versions in our monorepo.

Is there a known workaround? A way to trick parcel into assuming that `p/foo` is the root?

**Comment by app-akrobaten** [comment]
is there now a suitable solution? :)

**Comment by SantoJambit** [comment]
Nope, sadly, installing parcel in the monorepo root doesn't work either, as auto-installing will still fail, since yarn needs an extra parameter to install dependencies in the root.

I'm not on the project anymore (at work), which has the problem, and the new team is currently investigating other bundlers, which I am very sad about, but I can understand, that this is a deal-breaker :frowning:

**Comment by mischnic** [comment]
This should be fixed in the next nightly (= not yet but in about 12 hours) with https://github.com/parcel-bundler/parcel/pull/6414 (https://github.com/parcel-bundler/parcel/pull/6414/commits/64797ba4298362e183f11f6056a03db12842177e)

**Comment by jimmyhoran** [comment]
@mischnic thanks for looking into this original issue. I know the issue has been closed and expected to be resolved with the recent nightly, however, it seems unclear if this is fixed, and if so what version. I recently tried using the latest nightly version `2.0.0-nightly.734` and the issue persists for me.

```Failed to install @parcel/transformer-sass: Yarn failed to install modules:Child process failed```

Has anyone been able to validate that this original issue has been resolved by using one of the more recent nightly versions?

**Comment by jimmyhoran** [comment]
After looking into this a bit further I was able to resolve the issue using the `next` version tag, which is resolving to `2.0.0-beta.3.1`. For anyone else that stumbles upon this issue, I can confirm this is resolved for me with `2.0.0-beta.3.1`. Also, I found there's no need to use `--no-autoinstall` anymore and I was able to get hoisting of Parcel 2 with Yarn 2 successfully working within our mono-repo. Thanks @mischnic for all your work and support on this issue 👍
